File: SQL/Easy/268-missing-number/missing-number.java
import java.util.List;

class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int ans = -1;

        for (int i = 0; i <= n; i++) {
            boolean found = false;
            for (int num : nums) {
                if (num == i) {
                    found = true;
                    break;
                }
            }

            if (!found) {
                ans = i;
                break;
            }
        }

        if (!contains(nums, 0)) {
            return 0;
        }
        
        return ans;
    }

    private boolean contains(int[] nums, int target) {
        for (int num : nums) {
            if (num == target) {
                return true;
            }
        }
        return false;
    }
}

File: SQL/Easy/268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
         int n=nums.size();
        // if(n==1){
        //     if(nums[0]!=0)
        //         return 0;
        // }

        int ans;
        for(int i=1;i<=n;i++){
            auto it=i;
            if(find(nums.begin(),nums.end(),it)==nums.end()){
                ans=i;
                break;
            }
        }

        if(find(nums.begin(),nums.end(),0)==nums.end())
            return 0;
        return ans;
    }
};